substrate/substrate-runtime/src/metadata/runtime-description.ts

```diff
@@ -708,7 +721,14 @@ class FromOld {
                     docs: e.docs
                 }
             })
-            storage[mod.storage.prefix] = items
+            storage[mod.name] = {
+                prefix: mod.storage.prefix,
+                items
+            }
+
+            if (storage[mod.storage.prefix] == null) {
```

Comment:
This will generate extra items in "generate all" scenario and kinda pollutes metadata.

The same thing can be done within `Runtime` and typegen.

`private palletsByPrefix: {[prefix: string]: string}`